queries.sql

```diff
@@ -113,6 +113,69 @@ FROM species
 JOIN animals ON animals.species_id = species.id
 GROUP BY species.name;
 
+-- 1. Who was the last animal seen by William Tatcher?
+SELECT a.name AS animal_name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'William Tatcher'
+ORDER BY v.date DESC
+LIMIT 1;
+-- 2. How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT a.name) AS num_animals
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez';
+-- 3. List all vets and their specialties, including vets with no specialties.
+SELECT ve.name AS vet_name, GROUP_CONCAT(s.name SEPARATOR ', ') AS specialties
+FROM vets ve
+LEFT JOIN specializations sp ON ve.id = sp.vet_id
+LEFT JOIN species s ON sp.species_id = s.id
+GROUP BY ve.name;
+-- 4. List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name AS animal_name
+FROM animals a
+JOIN visits v ON a.id = v.animal_id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez' AND v.date BETWEEN '2020-04-01' AND '2020-08-30';
```

Comment:
- [ ] this query should return Agumon and Blossom according to the provided information, please double-check.